app/controllers/katello/concerns/api/v2/repository_content_controller.rb

```diff
@@ -63,7 +64,13 @@ def compare
         else
           repos = Katello::Repository.where(:content_view_version_id => @versions.pluck(:id))
           repos = repos.where(:root_id => @repo.root_id) if @repo
-          collection = filter_by_repos(repos, resource_class.all)
+          if (params[:restrict_comparision] == 'same')
+            collection = filter_same_by_repos(repos, resource_class.all, @versions.pluck(:id), true)
```

Comment:
No need to pluck versions individually.
```suggestion
            collection = filter_same_by_repos(repos, resource_class.all, @versions.select(:id), true)
```